apps/trackers/product_definition_handlers/base.py

```diff
@@ -18,44 +18,50 @@ def __init__(self) -> None:
             UnackedHandler(),
         ]
 
-    def file_tracker_offers(
+    def file_tracker_offer(
         self,
         affect: Affect,
         impact: Impact,
-        ps_module: PsModule,
+        ps_update_stream: PsUpdateStream,
         exclude_existing_trackers=False,
     ):
-        offers = {}
+        # Only generate offers for active streams
+        if not ps_update_stream.is_active:
+            return None
 
-        active_ps_update_streams = ps_module.active_ps_update_streams.all()
-        if exclude_existing_trackers:
-            active_ps_update_streams = active_ps_update_streams.exclude(
-                name__in=affect.trackers.values_list("ps_update_stream", flat=True)
-            )
+        if (
+            exclude_existing_trackers
+            and affect.tracker is not None
+            and affect.tracker.ps_update_stream == ps_update_stream.name
```

Comment:
Can there be a tracker linked to an affect but have a different PS update stream?

Reply:
I am thinking about it and I think the equality should simply be asserted by a validation. It is a matter of linking between affects and trackers so I would really consider affect with stream 1 linked to a tracker with stream 2 as seriously invalid data. It could have a bad consequences like that affect could seem fixed while the tracker was actually fixing a different stream etc.

I can imagine a situation when somebody modifies an affect stream when it has already a tracker linked. I think that the way the linking works would not result in a de-link. Therefore there should be hard blocking validation. If someone manually modifies the tracker and it syncs back, it should automatically de-link.

Reply:
We already have `_validate_tracker_affect_ps_update_stream` in `Tracker` which checks just that, so this might be redundant. I will study if this can be safely removed.